src/rpc/util.h

```diff
@@ -383,6 +400,36 @@ class RPCHelpMan
     RPCHelpMan(std::string name, std::string description, std::vector<RPCArg> args, RPCResults results, RPCExamples examples, RPCMethodImpl fun);
 
     UniValue HandleRequest(const JSONRPCRequest& request) const;
+    /**
+     * Helper to get a request argument or its default value.
+     * This function only works during m_fun(), i.e. it should only be used in RPC method implementations.
+     * Use Arg<Type> to get the argument or its default value.
+     * Use Arg<Type*> to get the argument or a falsy value.
```

Comment:
In commit "rpc: Add Arg() default helper" (fa56ad74160ba3c07a16a661fc14cded6deed9eb)

Haven't really looked at implementation yet, but can documentation be updated to say what happens with type mismatches? E.g. if arg is an int and you request a bool? Is the value just ignored and is a default returned instead? Is there an exception?

Also might be nice to add a short examples here showing what equivalent low level code is so developers can know how to use this helper and interpret it. Like `Arg<int>(3)` is equivalent to `request.params[3].isNumber() ? request.params[3].getInt<int>() : default_value` (or whatever the equivalent is)

Overall this seems like a nice improvement that should cut down on boilerplate. I am wondering if it might be possible later to extend this:

- to support arrays/objects
- to enforces only expected types are passed at compile time
- to enforce default not requested when there is no default at compile time
- to somehow be compatible with luke's #24963 which allows multiple types per parameter (if that is a good idea)

Reply:
> to enforces only expected types are passed at compile time

Do we want to revisit #22978 first before doing compile time type checking for RPC?

I think compile time RPC type checks would require templatising `RPCArg` by what is currently `m_type` (and `m_inner` perhaps), changing `RPCHelpMan` to be templatised based on a `std::tuple<...>` of `RPCArg<X>`s, and likewise for `RPCMethodImpl` which then gives the actual RPC implementation compile time access to its expected argument types.

I don't think you could reasonably access params by name and get compile time type checking; for the named-only params via `options`, you *could* turn `options` into an ordered tuple and access named-only params by their position from when you defined the `options` object.

For multitype parameters, I think it'd be best to just have a `Type::ANY` placeholder.

Reply:
> This sound like a bug in the source code, so I don't think it makes sense to document unreachable or unexpected code paths.

If it's a bug to to call the the Arg function with an unexpected type, it's important for the documentation to at say what types it is allowed to be called with, because I don't think there's another way of knowing this without a having a pretty deep understanding of the RPCHelpMan framework and digging into template and macro code.

If you just want the documentation to say "It is a bug to pass X type" instead of "Passing X type will throw an exception" that is fine of course.